core/modules/memman/MemFileSet.h

```diff
@@ -0,0 +1,120 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2016 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_QSERV_MEMMAN_MEMFILESET_H
+#define LSST_QSERV_MEMMAN_MEMFILESET_H
+
+// System headers
+#include <string>
+#include <unistd.h>
```

Comment:
Well, size_t is strictly for memory, file offset does not have anything to do with memory size (you should remember good old days when we had to read >2GB files on 32-bit OSes). For file offsets more portable type should be `off64_t` from `<sys/types.h>` but any 64-bit type should work ok for our purposes.